test/suites/telemetry/telemetry.robot

```diff
@@ -16,106 +17,109 @@ Test Tags           restart    slow
-${PROXY_PORT}                   ${EMPTY}
-${PROMETHEUS_HOST}              ${EMPTY}
-${PROMETHEUS_PORT}              ${EMPTY}
-${TELEMETRY_WRITE_ENDPOINT}     https://infogw.api.openshift.com/metrics/v1/receive
-${ENABLE_TELEMETRY}             SEPARATOR=\n
-...                             telemetry:
-...                             \ \ status: Enabled
-${JOURNAL_CURSOR}               ${EMPTY}
-${PULL_SECRET}                  /etc/crio/openshift-pull-secret
-${PULL_SECRET_METRICS}          /etc/crio/openshift-pull-secret-with-telemetry
-${PULL_SECRET_NO_METRICS}       /etc/crio/openshift-pull-secret-without-telemetry
+${PROXY_HOST}                           ${EMPTY}
+${PROXY_PORT}                           ${EMPTY}
+${PROMETHEUS_HOST}                      ${EMPTY}
+${PROMETHEUS_PORT}                      ${EMPTY}
+${TELEMETRY_WRITE_ENDPOINT}             https://infogw.api.openshift.com/metrics/v1/receive
+${PROXY_ENDPOINT}                       http://${PROXY_HOST}:${PROXY_PORT}/api/v1/write
+${ENABLE_TELEMETRY}                     SEPARATOR=\n
+...                                     telemetry:
+...                                     \ \ status: Enabled
+...                                     \ \ endpoint: ${TELEMETRY_WRITE_ENDPOINT}
+${DISABLE_TELEMETRY_TO_PROMETHEUS}      SEPARATOR=\n
+...                                     telemetry:
+...                                     \ \ status: Disabled
+...                                     \ \ endpoint: http://${PROMETHEUS_HOST}:${PROMETHEUS_PORT}/api/v1/write
+${ENABLE_TELEMETRY_WITH_PROXY}          SEPARATOR=\n
+...                                     telemetry:
+...                                     \ \ status: Enabled
+...                                     \ \ endpoint: ${TELEMETRY_WRITE_ENDPOINT}
+...                                     \ \ proxy: ${PROXY_ENDPOINT}
+${ENABLE_TELEMETRY_TO_PROMETHUS}        SEPARATOR=\n
+...                                     telemetry:
+...                                     \ \ status: Enabled
+...                                     \ \ endpoint: http://${PROMETHEUS_HOST}:${PROMETHEUS_PORT}/api/v1/write
+${JOURNAL_CURSOR}                       ${EMPTY}
+${PULL_SECRET}                          /etc/crio/openshift-pull-secret
+${PULL_SECRET_METRICS}                  /etc/crio/openshift-pull-secret-with-telemetry
+${PULL_SECRET_NO_METRICS}               /etc/crio/openshift-pull-secret-without-telemetry
 
 
 *** Test Cases ***
-MicroShift Reports Metrics To Server
+MicroShift Reports Metrics To Default Server
     [Documentation]    Check MicroShift is able to send metrics to the telemetry server without errors.
-    [Setup]    Setup Telemetry Configuration
+    [Setup]    Setup Telemetry Configuration    ${ENABLE_TELEMETRY}    ${PULL_SECRET_METRICS}
 
-    Wait Until Keyword Succeeds    10x    10s
-    ...    Should Find Metrics Success
+    Should Find Metrics In Journal Log Success    MicroShift telemetry starting, sending first metrics collection.
```

Comment:
I meant removing this line, as line 55 also ensures that this one is present. Mentioning this because relying on logs is usually not very robust, but for the next line we have no other option unless we implement yet more metrics in microshift (which we dont have, might be a good idea for the future).

Reply:
I agree, I'll keep the log message check as simple as possible. So I'll remove all code lines:
`Should Find Metrics In Journal Log Success    MicroShift telemetry starting, sending first metrics collection.`